### PR TITLE
CI: add GitHub Actions CI workflow

### PR DESCRIPTION
**Pull request by mdb**
👋 This adds a very basic MVP GitHub Actions-based CI workflow that runs all tests (via `go test ./...` from the project root) on each `git push` against the repository, as well as each PR opened against the repository.

It appears the tests currently fail (I can reproduce this locally as well); an example CI run can be viewed here, at my fork: https://github.com/mdb/go/actions/runs/5223060874. 

Is this helpful?

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (cdauphinee) [inline .github/workflows/ci.yaml]: Just `pull_request` by itself will re-run the action when someone pushes to a branch with an open PR, so no need for `push` unless you intended for it to run when branches that don't have an open PR are pushed to

Comment (mdb) [reply .github/workflows/ci.yaml]: @cdauphinee Yes; that was my original intention. However, I've pushed another change making it the following, which ensures we don't invoke redundant CI jobs when pushing to a branch for which there's an open PR.

```yaml
on:
  pull_request:
  push:
    branches:
      - master
```

Does that address your concern?

**Review by er9781 — APPROVED** [review]

**Comment by mdb** [comment]
@er9781 Thanks for the PR approval! What are next steps towards merging, given that I don't currently have merge permissions in this repo (should I have them?)? Do you recommend I get additional code review approvals?

File: .github/workflows/ci.yaml
Comment (cdauphinee) [reply .github/workflows/ci.yaml]: Got it, that makes sense!

**Pull request merged**
